File: lift/lift.rb
lift_a = 0
lift_b = 7

T = gets.chomp
test_cases = T.to_i

O = []

test_cases.times do
  floor = gets.chomp

  d_to_a = (floor.to_i - lift_a.to_i).abs
  d_to_b = (floor.to_i - lift_b.to_i).abs

  if d_to_a <= d_to_b
    lift_a = floor
    O.push 'A'
  else
    lift_b = floor
    O.push 'B'
  end
end

puts O

File: lift/lift.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var testCases, liftA, liftB, floor int = 0, 0, 7, 0
	fmt.Scanln(&testCases)

	output := make([]string, testCases)

	for i := 0; i < testCases; i++ {
		fmt.Scanln(&floor)
		distanceToA := math.Abs(float64(floor - liftA))
		distanceToB := math.Abs(float64(floor - liftB))

		if distanceToA <= distanceToB {
			liftA = floor
			output[i] = "A"
		} else {
			liftB = floor
			output[i] = "B"
		}
	}

	for _, o := range output {
		fmt.Println(o)
	}
}
